text/0097-log-data-model.md

```diff
@@ -0,0 +1,978 @@
+that emits the record or about the infrastructure where the application runs.
+Data formats that represent this data model may be designed in a manner that
+allows the `Resource` field to be recorded only once per batch of log records
+that come from the same source. SHOULD follow OpenTelemetry
+[semantic conventions for Resources](https://github.com/open-telemetry/opentelemetry-specification/tree/master/specification/resource/semantic_conventions).
+
+### Field: `Attributes`
+
+Type: key/value pair list.
+
+Description: Additional information about the specific event occurrence. "value"
+of each pair is of `any` type. Unlike the `Resource` field, which is fixed for a
+particular source, `Attributes` can vary for each occurrence of the event coming
+from the same source. Can contain information about the request context (other
+than TraceId/SpanId). SHOULD follow OpenTelemetry
+[semantic conventions for Attributes](https://github.com/open-telemetry/opentelemetry-specification/tree/master/specification/trace/semantic_conventions).
+
+## Example Log Records
+
+Below are examples that show one possible representation of log records in JSON.
+These are just examples to help understand the data model. Don’t treat the
+examples as _the_ way to represent this data model in JSON.
+
+This document does not define the actual encoding and format of the log record
+representation. Format definitions will be done in separate OTEPs (e.g the log
+records may be represented as msgpack, JSON, Protocol Buffer messages, etc).
+
+Example 1
+
+```javascript
+{
+  "Timestamp": 1586960586000, // JSON needs to make a decision about 
+                              // how to represent nanoseconds.
+  "Attributes": {
+    "http.status_code": 500,
+    "http.url": "http://example.com",
+    "my.custom.application.tag": "hello",
+  },
+  "Resource": {
+    "service.name": "donut_shop",
+    "service.version": "semver:2.0.0",
+    "k8s.pod.uid": "1138528c-c36e-11e9-a1a7-42010a800198",
+  },
+  "TraceId": "f4dbb3edd765f620", // this is a byte sequence 
+                                  // (hex-encoded in JSON)
+  "SpanId": "43222c2d51a7abe3",
+  "SeverityText": "INFO",
+  "SeverityNumber": 9,
+  "Body": "20200415T072306-0700 INFO I like donuts"
+}
+```
+
+Example 2
+
+```javascript
+{
+  "Timestamp": 1586960586000,
+  ...
+  "Body": {
+    "i": "am",
+    "an": "event",
+    "of": {
+      "some": "complexity"
+    }
+  }
+}
+```
+
+## Open Questions
+
+- Should we store entire
+  [W3C Trace Context](https://www.w3.org/TR/trace-context/), including
+  `traceparent` and `tracestate` fields instead of only `TraceFlags`?
+  
+- Is `SeverityText`/`SeverityNumber` fields design good enough?
+
+- Early draft of this proposal specified that `Timestamp` should be populated
+  from a monotonic, NTP-synchronized source. I removed this requirement to avoid
+  confusion. Do we need any requirements for timestamp sources?
+
+- Is there a need for special treatment of security logs?
```

Comment:
My opinion: not in this data model. That would be a concern for other aspects of logging:

1. Where is the data stored?
2. How is the data transmitted (wire protocols)?
3. How does security data emerge from the observed system?

Reply:
Agreed. Resolving. Feel free to reopen if needed.